libbeat/publisher/queue/queuetest/queuetest.go

```diff
@@ -209,12 +214,29 @@ func runTestCases(t *testing.T, tests []testCase, queueFactory QueueFactory) {
 
 			go test.producers(&wg, nil, log, queue)()
 			go test.consumers(&wg, nil, log, queue)()
-
+			go testFetchMetrics(t, queue)
 			wg.Wait()
 		}))
 	}
 }
 
+func testFetchMetrics(t *testing.T, mon queue.Queue) {
+	_, err := mon.Metrics()
+	if errors.Is(err, queue.ErrMetricsNotImplemented) {
+		return
+	}
+
+	for {
+		metrics, err := mon.Metrics()
+		if errors.Is(err, io.EOF) {
+			continue
+		}
+		t.Logf("Got event count: %d/%d", metrics.EventCount.ValueOr(0), metrics.ByteCount.ValueOr(0))
+		assert.True(t, metrics.EventCount.Exists() || metrics.ByteCount.Exists())
+		time.Sleep(time.Second)
```

Comment:
Why do we need this sleep, and what does this need to be in a for loop? This seems like a bit of an odd test in that it doesn't exit once it has verified what it needs to but I'm not very familiar with the queue tests.

Reply:
Yah, it's an extremely odd test, and I wrote it while I was still figuring out how the weird custom test setup worked. I just wanted to have some kind of test when I put in the first commit.

Reply:
Alright, added a proper test.